src/all/nijiero/src/eu/kanade/tachiyomi/extension/all/nijiero/Nijiero.kt

```diff
@@ -0,0 +1,148 @@
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Nijiero : ParsedHttpSource() {
+    override val name = "Nijiero"
+    override val lang = "all"
+    override val supportsLatest = true
+
+    override val baseUrl = "https://www.nijiero-ch.com"
+
+    override val client = network.client.newBuilder().followRedirects(true).build()
+	
+    class TagFilter : Filter.Select<String>("Category", nijieroTags)
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select("#entry > ul > li a[href]").mapIndexed { i, linkElement ->
+            val linkUrl = linkElement.attr("href").removeSuffix(".webp").removePrefix(baseUrl)
+            Page(i, document.location(), linkUrl)
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
+	
+    private val spaceRegex = Regex("""\s+""")
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val tagIdx: Int = (filters.last() as TagFilter).state
+
+        val keyword = when {
+            query.isBlank() -> nijieroTags[tagIdx]
+            else -> query
+        }.replace(spaceRegex, "-").lowercase()
+
+        val uniqueParam = System.currentTimeMillis()
+        var url = baseUrl.toHttpUrl().newBuilder()
+            .addEncodedPathSegment("category")
+            .addEncodedPathSegment(keyword)
+            .addEncodedPathSegment("page")
+            .addEncodedPathSegment(page.toString())
+            .addQueryParameter("refresh", uniqueParam.toString())
+            .build()
+		
+        // if 404, recheck
+        if (client.newCall(GET(url, headers)).execute().code == 404) {
+            url = baseUrl.toHttpUrl().newBuilder()
+                .addEncodedPathSegment("tag")
+                .addEncodedPathSegment(keyword)
+                .addEncodedPathSegment("page")
+                .addEncodedPathSegment(page.toString())
+                .addQueryParameter("refresh", uniqueParam.toString())
+                .build()
+            // if there is a better solution just *tell me* .
+        }
+
+        return GET(url, headers)
+    }
+    override fun searchMangaNextPageSelector() = ".next.page-numbers"
+    override fun searchMangaSelector() = ".contentList > div:has(a)"
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        element.select("a")?.let { link ->
```

Comment:
```suggestion
        element.select("a").let { link ->
```
`select` isn't nullable